workflow/controller/controller.go

```diff
@@ -973,6 +1019,22 @@ func (wfc *WorkflowController) addWorkflowInformerHandlers(ctx context.Context)
 				DeleteFunc: func(obj interface{}) {
 					// IndexerInformer uses a delta queue, therefore for deletes we have to use this
 					// key function.
+
+					// Remove finalizers if they exist before deletion
+					pods := wfc.kubeclientset.CoreV1().Pods(wfc.GetManagedNamespace())
+					podList, err := pods.List(ctx, metav1.ListOptions{
+						LabelSelector: fmt.Sprintf("%s=%s", common.LabelKeyWorkflow, obj.(*unstructured.Unstructured).GetName()),
+					})
+					if err != nil {
+						log.WithError(err).Error("Failed to list pods")
+					}
+					for _, p := range podList.Items {
+						patch := createFinalizerPodStatusRemovalPatchIfExists(&p)
+						if err := applyPatchIfExists(ctx, pods, p.Name, patch); err != nil {
+							log.WithError(err).Error("Failed to remove finalizer from pod")
+						}
```

Comment:
Can I understand why we do it here for the case of Workflow having been deleted? I presume we will call the `deletePod` logic anyway (although haven't yet looked where).?

Reply:
or is this perhaps the case where the OwnerReference was automatically deleting the Pods remaining?

Reply:
I had assumed this code had already been reviewed on the original PR, so this implementation simply followed it.
But I'm also not sure why we're doing this here.
In fact, tests are successful without this implementation.
If you also believe that this implementation is unnecessary, I would like to remove it.

Reply:
sounds good!

Reply:
I have removed this in the commit below.
https://github.com/argoproj/argo-workflows/pull/12413/commits/de2bc6ce34c6cae6f40a204686fca0b2fcd604c9